sympy/matrices/matrices.py

```diff
@@ -186,48 +186,61 @@ def __mul__(self, other):
     def __neg__(self):
         return -1*self
 
-    def __pow__(self, num):
-        from sympy.matrices import eye, diag, MutableMatrix
+    def _matrix_pow_by_recursion(self, num):
+        from sympy.matrices import eye
+        n = int(num)
+        if n < 0:
+            return self.inv()**-n   # A**-2 = (A**-1)**2
+        a = eye(self.cols)
+        s = self
+        while n:
+            if n % 2:
+                a *= s
+                n -= 1
+            if not n:
+                break
+            s *= s
+            n //= 2
+        return self._new(a)
+
+    def _matrix_pow_by_jordan_blocks(self, num):
+        from sympy.matrices import diag, MutableMatrix
         from sympy import binomial
```

Comment:
I've defined it under class MatrixBase with 4 indentation space. could you please elaborate whats wrong in it.